dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/CurrentUserService.java

```diff
@@ -55,14 +55,6 @@
      */
     User getCurrentUser();
 
-    User getCurrentUserInTransaction();
-
-    /**
-     * @return the user info for the currently logged in user. If no user is
-     *         logged in or the auto access admin is active, null is returned.
-     */
-    UserInfo getCurrentUserInfo();
```

Comment:
`getCurrentUserInfo();` has been introduced to resolve the performance issue of `getCurrentUser()` getting called too many times, as it queries to database and became very slow.
we might need to keep `userInfo` object to avoid any regression performance issue.
Otherwise we may need to carefully test the performance of `getCurrentUser();`

Reply:
Yes, my intention was to get started with the refactor of getCurrentUser() as soon as this is merged. The intended refactor will put the "current user" object into memory only and avoid all together the DB lookups after authentication. I expect that to be as fast as the previous.